src/main/java/user_code_resolver/UserCodeResolver.java

```diff
@@ -0,0 +1,263 @@
+        //////// Get all elementValueChanges that are of Type ListChanges and update the project based on them ////////
+        for (Change change : diff.getChanges()) {
+            if (change instanceof ListChange listChange) {
+                logger.info("The following structural changes have been found:");
+                for (ContainerElementChange element : listChange.getChanges()) {
+                    if (element instanceof ElementValueChange elementValueChange) {
+                        logger.info(elementValueChange);
+                        project = updateFilesBasedOnElementValueChange(elementValueChange, project);
+                    }
+                }
+            }
+        }
+        return project;
+    }
+
+    /**
+     * Converts the file list to a string list
+     *
+     * @param fileList
+     * @return fileList converted to a string list
+     */
+    private List<String> convertNewProjectStructureToList(List<File> fileList) {
+        return fileList.stream().map(String::valueOf).collect(Collectors.toList());
+    }
+
+    /**
+     * This method updates the Impl files based on the ElementValueChange Object
+     *
+     * @param elementValueChange
+     * @param project
+     * @return updated project object
+     */
+    private Project updateFilesBasedOnElementValueChange(ElementValueChange elementValueChange, Project project) {
+        //////// Get all relevant values from the ElementValue Change object ////////
+        String oldPath = elementValueChange.getLeftValue().toString();
+        String newPath = elementValueChange.getRightValue().toString();
+        String[] oldPathArray = oldPath.split("\\\\");
+        String[] newPathArray = newPath.split("\\\\");
+        int fileTypeStart = oldPathArray[oldPathArray.length - 1].indexOf(".");
+        String oldConstructorName = oldPathArray[oldPathArray.length - 1].substring(0, fileTypeStart);
+        String newConstructorName = newPathArray[oldPathArray.length - 1].substring(0, fileTypeStart);
+
+        //////// Find and update the relevant file with the new project  ////////
+        for (UserFile file : project.getFiles()) {
+            if (file.getFilename() == elementValueChange.getLeftValue()) {
+                file.setFilename(newPath);
+                for (int i = 0; i < file.getContent().size(); i++) {
+                    String line = file.getContent().get(i);
+                    if (line == null) continue;
+                    if (line.contains("package")) {
+                        line = getNewPackageName(oldPathArray);
+                        file.getContent().set(i, line);
+                    }
+                    if (line.contains("class ") || line.contains(oldConstructorName)) {
+                        file.getContent().set(i, line.replace(oldConstructorName, newConstructorName));
+                    }
+                }
+                //TODO check if elementVlaueChange only has 1 value if yes add break;
```

Comment:
ToDo?!?

Reply:
Ja das müsste ich mal mit zwei unterschiedlichen xml Projekt Strukturen testen. give me and i can do it

Reply:
Wo soll ich die den herzaubern. Machse dir doch selber :D